### PR TITLE
Allowing openblas to be build on a VM

### PR DESCRIPTION
**Pull request by briedel**

### CODE REVIEW

File: var/spack/repos/builtin/packages/openblas/package.py [hunk 1]
```diff
@@ -61,6 +62,18 @@ class Openblas(MakefilePackage):
...
     )
 
+    variant(
+        'virtual_machine',
+        default=False,
+        description="Adding options to build openblas on Linux virtual machine"
+    )
+
+    variant(
+        'num_threads',
```
Comment (davydden) [inline var/spack/repos/builtin/packages/openblas/package.py]: maybe it's better to control this via environment variables, i.e.
```
export OPENBLAS_NUM_THREADS=4
```
?

p.s. Most blas/lapack in spack have `threads` variant to control which multi-threading to use (if use anything at all).

File: var/spack/repos/builtin/packages/root/package.py [hunk 2]
```diff
@@ -55,6 +55,7 @@ class Root(CMakePackage):
     depends_on('cmake@3.4.3:', type='build')
     depends_on('pkg-config',   type='build')
 
+    depends_on('binutils')
```
Comment (davydden) [inline var/spack/repos/builtin/packages/root/package.py]: this does not seem to be related to the PR.

Comment (briedel) [reply var/spack/repos/builtin/packages/root/package.py]: That should have already made it into the repo. Weird

File: var/spack/repos/builtin/packages/openblas/package.py (hunk 1 above)
Comment (briedel) [reply var/spack/repos/builtin/packages/openblas/package.py]: Okay I can take it out. In my handcrafted system, I was defining it at compile time.

**Comment by briedel** [comment]
See 535bee5 for requested changes

**Comment by briedel** [comment]
Removed ROOT changes after merge with main repo.

**Comment by scheibelp** [comment]
Thanks!
